Apps/Src/UpdateVelocity.c

```diff
@@ -48,23 +53,56 @@ void Task_UpdateVelocity(void *p_arg)
         uint8_t accelPedalPercent = Pedals_Read(ACCELERATOR);
         uint8_t brakePedalPercent = Pedals_Read(BRAKE);
 
+        //Edge detector to toggle the cruise enable on only the rising edge
+        State currEnable = Switches_Read(CRUZ_EN);
+        if((prevEnable == OFF) && (currEnable == ON)){ //we went from off to on, so we should toggle it
+            prevEnable = ON;
+            cruiseEnable ^= 1; 
+        } else if ((currEnable == OFF)){ //went from on to OFF, so reset the prev state
+            prevEnable = OFF;
+        }
+
+        //Edge detector to toggle the cruise set on only the rising edge
+        State currSet = Switches_Read(CRUZ_ST);
+        if((prevSet == OFF) && (currSet == OFF)){
+            prevSet = ON;
+            cruiseSet ^= 1;
+        } else if ((currSet == OFF)){
+            prevSet = OFF;
+        }
+
+
+        State cruiseState = cruiseSet & cruiseEnable; //if both are true, then we are in cruise mode, if not, we are in normal drive
+        
+        //record speed when we're not already cruising
+        if(!cruiseState){
+            cruiseSpeed = MotorController_ReadVelocity();
+        }
+
         // Set brake lights
         if(brakePedalPercent >= UNTOUCH_PEDALS_PERCENT){
             Lights_Set(BrakeLight, ON);
         }
         else{
             Lights_Set(BrakeLight, OFF);
         }
-    
+        
+
         // Deadband comparison
         if(brakePedalPercent >= UNTOUCH_PEDALS_PERCENT){
```

Comment:
Honestly, this is just from a mechnical perspective. It might be nice to make untouch pedals percent a bit bigger since it will be really hard to not press that pedal when sitting in the car. Just to put it out there.

Reply:
Yeah, from a software perspective, brake is treated just like a button, so it's just pressed/unpressed. I am concerned that if we make untouch a bigger value, we will also reduce the one-pedal drive regen range. @ClarkPoon What do you think?

Reply:
One idea is that we could boost both the untouch and the neutral pedal point for one pedal drive